versioned/tests/src/main/java/org/projectnessie/versioned/tests/AbstractNamespaceValidation.java

```diff
@@ -509,4 +510,42 @@ void deleteHierarchy() throws Exception {
                             .collect(Collectors.toList())))
         .doesNotThrowAnyException();
   }
+
+  @Test
+  void renameWithNonExistingNamespace() throws Exception {
+    BranchName branch = BranchName.of("renameWithNonExistingNamespace");
+    store().create(branch, Optional.empty());
+
+    Namespace ns1 = Namespace.of("ns");
```

Comment:
I think you cold simplify the test, we don't need 2 namespaces:

```java
    Namespace ns = Namespace.of("non_existing");
    ContentKey key1 = ContentKey.of("table");
    ContentKey key2 = ContentKey.of(ns, "table");

    store()
        .commit(
            branch,
            Optional.empty(),
            fromMessage("create a table"),
            singletonList(Put.of(key1, newOnRef("value"))));

    Content table = store().getValue(branch, key1).content();
// etc.
```

Reply:
Done. Also added an entry for CHANGELOG.md.